### PR TITLE
Fix #347 - add bramble.export() for creating project.zip, an archive of entire fs

### LINKED ISSUE #347: Support zip export
I want to add an extension that adds zip and unzip commands to the `CommandManager`, with the goal of being able to unzip a `.zip` archive into your fs, and to export your fs to a `.zip`.  Prior art in:
* https://github.com/filerjs/filer/blob/08f4b7e528b8891feec301f549ec9e58a7496304/src/shell/shell.js#L479-608 + https://github.com/Stuk/jszip
* https://github.com/humphd/nohost/blob/master/src/webserver.js#L63-L84

### PR DESCRIPTION
**Pull request by humphd**
This adds a new `bramble.export()` API call, which zips the project's root dir downward, and creates a `project.zip` file with a `project/*` dir, containing all the project files.

@Pomax was asking for this today on Twitter, https://twitter.com/TheRealPomax/status/630099230061936644.  So here it is!  @Pomax, you can test this by opening your console, and typing `bramble.export()`.  I'll add some proper UI in Thimble next week in a patch there.

### CODE REVIEW

**Comment by Pomax** [comment]
That looks to be working pretty well! Although the layout I get hints at something going wrong at the file system level, perhaps:

```
project.zip
|- project
|   |- ...my files...
\-- 7
     \- projects
         \- 30
            \- <no content in this dir>
```

**Comment by humphd** [comment]
Nice catch, @Pomax, I've pushed a new version oft his, which fixes that bug.  I needed a regex vs. string matching to catch the case of a missing trailing slash on the project root dir itself.  Fixed!

**Comment by Pomax** [comment]
R+, that works great!

File: src/filesystem/impls/filer/ArchiveUtils.js [hunk 1]
```diff
@@ -141,39 +142,23 @@ define(function (require, exports, module) {
...
 
         function addFile(path, callback) {
-            fs.readFile(path, function(err, data) {
+            fs.readFile(path, {encoding: null}, function(err, data) {
                 if(err) {
                     return callback(err);
                 }
 
-                archive.file(toRelPath(path), data.buffer, {binary: true});
+                jszip.file(toRelProjectPath(path), data.buffer, {binary: true});
```
Comment (sedge) [inline src/filesystem/impls/filer/ArchiveUtils.js]: Should this be `JSZip.file`?

File: src/filesystem/impls/filer/ArchiveUtils.js [hunk 2]
```diff
@@ -181,11 +166,7 @@ define(function (require, exports, module) {
         function addDir(path, callback) {
             fs.readdir(path, function(err, list) {
                 // Add the directory itself
-                archive.folder(toRelPath(path));
-
-                if(!options.recursive) {
-                    return callback();
-                }
+                jszip.folder(toRelProjectPath(path));
```
Comment (sedge) [inline src/filesystem/impls/filer/ArchiveUtils.js]: ditto

File: src/filesystem/impls/filer/ArchiveUtils.js [hunk 3]
```diff
@@ -210,22 +191,18 @@ define(function (require, exports, module) {
             });
         }
 
-        var archive = new JSZip();
+        // TODO: we should try to move this to a worker
+        var jszip = new JSZip();
```
Comment (sedge) [inline src/filesystem/impls/filer/ArchiveUtils.js]: Ah, or not. Shouldn't this be declared earlier?

Comment (humphd) [inline src/filesystem/impls/filer/ArchiveUtils.js]: It gets hoisted, I'm not sure it matters.

File: src/filesystem/impls/filer/ArchiveUtils.js [hunk 4]
```diff
@@ -210,22 +191,18 @@ define(function (require, exports, module) {
...
 
-        // Make sure the zipfile doesn't already exist.
-        fs.exists(zipfile, function(exists) {
-            if(exists) {
-                return callback(new Error('zipfile already exists', zipfile));
+        async.eachSeries([root], add, function(err) {
```
Comment (sedge) [inline src/filesystem/impls/filer/ArchiveUtils.js]: This is only going to run once anyway isn't it? Why bother with `eachSeries`?

Comment (humphd) [inline src/filesystem/impls/filer/ArchiveUtils.js]: Good point.  I used to let you pass in arbitrary paths.  I'll fix.

**Comment by sedge** [comment]
r+ with async fix

**Pull request merged**